node/broadcaster/process.go

```diff
@@ -20,85 +20,58 @@ func (b Broadcaster) GetHeight() int64 {
+		pendingTxTime := time.Unix(0, pendingTx.Timestamp)
+
+		// before timeout
+		if lastBlockResult.Block.Time.Before(pendingTxTime.Add(b.cfg.TxTimeout)) {
+			b.logger.Debug("failed to query tx", zap.String("tx_hash", pendingTx.TxHash), zap.String("error", txerr.Error()))
+			return nil, time.Time{}, types.ErrTxNotFound
+		} else {
+			// timeout case
+			account, err := b.GetAccount(b.getClientCtx(ctx), b.keyAddress)
 			if err != nil {
-				return err
+				return nil, time.Time{}, err
 			}
-		}
-	}
 
-	// check timeout of pending txs
-	// @sh-cha: should we rebroadcast pending txs? or rasing monitoring alert?
-	if length := b.LenLocalPendingTx(); length > 0 {
-		b.logger.Debug("remaining pending txs", zap.Int64("height", block.Block.Height), zap.Int("count", length))
-		pendingTxTime := time.Unix(0, b.peekLocalPendingTx().Timestamp)
-		if block.Block.Time.After(pendingTxTime.Add(b.cfg.TxTimeout)) {
-			panic(fmt.Errorf("something wrong, pending txs are not processed for a long time; current block time: %s, pending tx processing time: %s", block.Block.Time.UTC().String(), pendingTxTime.UTC().String()))
+			// if sequence is larger than the sequence of the pending tx,
+			// handle it as the tx has already been processed
+			if pendingTx.Sequence < account.GetSequence() {
+				return nil, time.Time{}, nil
+			}
+			panic(fmt.Errorf("something wrong, pending txs are not processed for a long time; current block time: %s, pending tx processing time: %s", time.Now().UTC().String(), pendingTxTime.UTC().String()))
 		}
 	}
 
-	// update last processed block height
-	b.lastProcessedBlockHeight = latestChainHeight
-
-	return nil
-}
-
-// CheckPendingTx query tx info to check if pending tx is processed.
-func (b *Broadcaster) CheckPendingTx(ctx context.Context) (*btypes.PendingTxInfo, *rpccoretypes.ResultTx, time.Time, error) {
-	if b.LenLocalPendingTx() == 0 {
-		return nil, nil, time.Time{}, nil
-	}
-
-	pendingTx := b.peekLocalPendingTx()
-	pendingTxTime := time.Unix(0, b.peekLocalPendingTx().Timestamp)
-
-	lastBlockResult, err := b.rpcClient.Block(ctx, nil)
-	if err != nil {
-		return nil, nil, time.Time{}, err
-	}
-	if lastBlockResult.Block.Time.After(pendingTxTime.Add(b.cfg.TxTimeout)) {
-		// @sh-cha: should we rebroadcast pending txs? or rasing monitoring alert?
-		panic(fmt.Errorf("something wrong, pending txs are not processed for a long time; current block time: %s, pending tx processing time: %s", time.Now().UTC().String(), pendingTxTime.UTC().String()))
-	}
-
-	txHash, err := hex.DecodeString(pendingTx.TxHash)
-	if err != nil {
-		return nil, nil, time.Time{}, err
-	}
-	res, err := b.rpcClient.QueryTx(ctx, txHash)
-	if err != nil {
-		b.logger.Debug("failed to query tx", zap.String("tx_hash", pendingTx.TxHash), zap.String("error", err.Error()))
-		return nil, nil, time.Time{}, nil
-	}
-
 	blockResult, err := b.rpcClient.Block(ctx, &res.Height)
```

Comment:
https://b545809c-5562-4e60-b5a1-22e83df57748.initiation-2.mesa-rpc.newmetric.xyz/header?height=100
block header query should be better to get the time of block in a cheap cost